backend/entityservice/tests/util.py

```diff
@@ -196,17 +196,27 @@ def create_project_upload_data(
     return new_project_data, json_responses
 
 
+def _check_delete_response(r):
+    # Note we allow for a 403 because the project may already have been deleted
```

Comment:
Just wondering: why not 404? Not part at all with this PR as you haven't changed anything about it, but I'm seeing in the swagger file that we can return a 403 or a 404. Now, if a run has been deleted, I would assume this is not available, so we should return a 404, but as we are also requesting access token, and a run does not have any if not present, should return indeed a 403. So the question would be: is there any cases where we could return a 404 on the `/runs` endpoints?

Reply:
This was to prevent enumeration on the `/runs` endpoint - overkill with our large UUID addresses. I think we only use 404 for wrong address e.g. `/links` instead of `/runs`